server/routes/api/place.py

```diff
@@ -763,3 +766,124 @@ def placeid2dcid():
                 result[inId] = dcid
         return Response(json.dumps(result), 200, mimetype='application/json')
     abort(404, 'no valid dcids found for %s' % place_ids)
+
+
+@cache.cached(timeout=3600 * 24, query_string=True)  # Cache for one day.
+@bp.route('/ranking_chart/<path:dcid>')
+def api_ranking_chart(dcid):
+    """
+    Gets the ranking data (including ranks, placename, value, date, and source) for a given place.
+
+    Return value example:
+    {
+        "statvar":{
+            "date":"2022-03",
+            "data":[
+                {
+                    "rank":1,
+                    "value":0.1,
+                    "placeDcid":"geoId/06",
+                    "placename":"California"
+                }
+            ],
+            "numDataPoints":1,
+            "exploreUrl":"/ranking/UnemploymentRate_Person_Rural/State/country/USA?h=geoId/06&unit=%",
+            "sources":[
+                "https://www.bls.gov/lau/"
+            ]
+        }
+    }
+    """
+    result = {}
+    isEarth = dcid == "Earth"
+    # Get the parent place.
+    if not isEarth:
+        list_of_parent_place = get_parent_place(dcid).get(dcid, [])
+        # Get the last parent place returned.
+        if list_of_parent_place:
+            parent_place_dcid = list_of_parent_place[-1]["dcid"]
+            place_type = get_place_type(dcid)
+        else:
+            flask.abort(500)
+    else:
+        parent_place_dcid = "Earth"
+        place_type = "Country"
+    configs = get_ranking_chart_configs()
+    # Get the first stat var of each config.
+    stat_vars, _ = shared_api.get_stat_vars(configs)
+    sv_data = dc.get_stat_set_within_place(parent_place_dcid, place_type,
+                                           list(stat_vars), "")
+    sv_data_values = sv_data.get("data", {})
+    if not sv_data or not sv_data_values:
+        flask.abort(500)
+    sv_metadata = sv_data.get("metadata", {})
+    # Consider the configs with single sv but ignore denominators.
+    for config in configs:
+        sv = config["statsVars"][0]
+        sv_data_stat = sv_data_values.get(sv, {}).get("stat", {})
+        if not sv_data_stat:
+            continue
+        sources = set()
+        dates = set()
+        data_points = []
+        for place_dcid in sv_data_stat:
+            # dcid_data is a dictionary including date, value and metahash.
+            dcid_data = sv_data_stat[place_dcid]
+            dcid_value = dcid_data.get("value", None)
+            dcid_place_name = get_name([place_dcid]).get(place_dcid, "")
```

Comment:
instead of making the get_name call for every place for every config, it'll be better to make the call once for all the places you will need names for, save the result in a variable, and just read that variable. 

This can be done before looping through the configs and the list of places that need names can be found by looking through sv_data_values

Reply:
Good point! Done